api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -255,6 +259,68 @@ private void deleteUserRecentlyModifiedResourceEntry(
     }
   }
 
+  @Override
+  public List<DbUserRecentlyModifiedResource> findAllRecentlyModifiedResourcesByUser(long userId) {
+    try {
+      return DbRetryUtils.executeAndRetry(
+          () -> populateRecentlyModifiedResources(userId), Duration.ofSeconds(1), 5);
+    } catch (InterruptedException e) {
+      throw new ServerErrorException(
+          "Unable to find Recently Modified Resources for user" + userId);
+    }
+  }
+
+  public List<DbUserRecentlyModifiedResource> populateRecentlyModifiedResources(long userId) {
+    List<DbUserRecentlyModifiedResource> userList =
+        userRecentlyModifiedResourceDao.findDbUserRecentResourcesByUserId(userId);
+    userList.stream()
+        .forEach(
+            userRecentResource -> {
+              switch (userRecentResource.getResourceType()) {
+                case COHORT:
```

Comment:
what about cohort review?

Reply:
I was looking at the original document for recent resource, https://docs.google.com/document/d/1Fd00DPfvPwzOfwkLlpQpjCVxlow1yQ1uSnlFUrxm_gY/edit#
It says "Cohorts: cache on createCohort, call logRecentResources when opening a cohort review "
I have not gotten an answer yet should we extend cohort review save/update as COHORT or COHORT_REVIEW

Reply:
ok thanks